src/wallet/bdb.cpp

```diff
@@ -656,16 +657,20 @@ void BerkeleyDatabase::ReloadDbEnv()
     env->ReloadDbEnv();
 }
 
-bool BerkeleyBatch::StartCursor()
+BerkeleyCursor::BerkeleyCursor(BerkeleyDatabase& database)
 {
-    assert(!m_cursor);
-    if (!pdb)
-        return false;
-    int ret = pdb->cursor(nullptr, &m_cursor, 0);
-    return ret == 0;
+    if (!database.m_db.get()) {
+        throw std::runtime_error(STR_INTERNAL_BUG("BerkeleyDatabase does not exist"));
+    }
+    assert(database.m_db.get());
+    int ret = database.m_db->cursor(nullptr, &m_cursor, 0);
+    assert(ret == 0);
```

Comment:
The assert is still there, making the code below unreachable.

Reply:
Oops, fixed.